src/OpenFeature.Contrib.Providers.Flagd/FlagdProvider.cs

```diff
@@ -22,44 +22,44 @@ namespace OpenFeature.Contrib.Providers.Flagd
     /// </summary>
     public sealed class FlagdProvider : FeatureProvider
     {
+        static int EventStreamRetryBaseBackoff = 1;
+        private readonly FlagdConfig _config;
         private readonly Service.ServiceClient _client;
         private readonly Metadata _providerMetadata = new Metadata("flagd Provider");
 
+        private readonly ICache<string, ResolutionDetails<Value>> _cache;
+        private int _eventStreamRetries;
+        private int _eventStreamRetryBackoff = EventStreamRetryBaseBackoff;
+
+        private System.Threading.Mutex _mtx;
+
         /// <summary>
         ///     Constructor of the provider. This constructor uses the value of the following
         ///     environment variables to initialise its client:
-        ///     FLAGD_HOST         - The host name of the flagd server (default="localhost")
-        ///     FLAGD_PORT         - The port of the flagd server (default="8013")
-        ///     FLAGD_TLS          - Determines whether to use https or not (default="false")
-        ///     FLAGD_SOCKET_PATH - Path to the unix socket (default="")
+        ///     FLAGD_HOST                     - The host name of the flagd server (default="localhost")
+        ///     FLAGD_PORT                     - The port of the flagd server (default="8013")
+        ///     FLAGD_TLS                      - Determines whether to use https or not (default="false")
+        ///     FLAGD_SOCKET_PATH              - Path to the unix socket (default="")
+        ///     FLAGD_CACHE                    - Enable or disable the cache (default="false")
+        ///     FLAGD_MAX_CACHE_SIZE           - The maximum size of the cache (default="10")
+        ///     FLAGD_MAX_EVENT_STREAM_RETRIES - The maximum amount of retries for establishing the EventStream
```

Comment:
The env stuff is important, but I think with all these additions, we _may_ want a constructor that also takes the config object directly.

Doesn't have to be part of this PR.

Reply:
Agree, that would certainly make sense. If it's ok, I would do this in a follow up PR

Reply:
yep!